library/std/src/io/mod.rs

```diff
@@ -1770,6 +1770,9 @@ pub trait Seek {
     /// Rewind to the beginning of a stream.
     ///
     /// This is a convenience method, equivalent to `seek(SeekFrom::Start(0))`.
+    /// Note that unlike `seek(SeekFrom::Start(0))`, this method does not return
+    /// the old offset and cannot replace assignements of the form:
+    /// `let old_offset = buf.seek(SeekFrom::Start(0))?;`.
```

Comment:
`seek` doesn't return the old position unless you do `SeekFrom::Current(0)`

Reply:
See also [the documentation of seek](https://doc.rust-lang.org/stable/std/io/trait.Seek.html#tymethod.seek):

> If the seek operation completed successfully, this method returns the **new** position from the start of the stream.